blog/content/post/2020-11-30-coreboot_for_power.md

```diff
@@ -0,0 +1,154 @@
+partitions, summing up to just below 32 megabytes. It runs from cache until it
+trains main memory, so it is limited to just 10 MB (size of L3 cache) split
+between two cores. To make it possible to load and execute that amount of code,
+[Hostboot uses on-demand paging](https://youtu.be/fTLsS_QZ8us?t=1559). It is
+similar to swap used by operating systems, except that it uses ROM instead of
+writable media, so only code and constant data can be discarded from memory (L3
+cache in this case).
+
+Another issue was nicely formulated by Timothy Pearson from Raptor CS:
+
+> Hostboot is a VM to run FSI routines, and the FSI routines are written one by
+> one by the hardware engineers designing the silicon itself.  This leads to
+> overly complex and difficult to understand code, as you can see.
+
+FSI is one of the buses Hostboot has to initialise. By simplifying the code we
+can make it **easier to understand** (no need to translate virtual addresses to
+the physical ones), **faster** (by discarding a dozen of function calls before
+the proper write to the register happens) and **smaller**. Smaller code means
+less time spent on reading from flash, but also more space for other components,
+like new drivers for Skiroot.
+
+#### For the coreboot community
+
+coreboot would get a **new architecture** supported. Although there are stubs,
+cross-compiler and libraries for PPC, the coreboot itself isn't in working state
+when it comes to this architecture as of now. It would also get a **new, truly
+open** ([unless you want SATA controller](https://wiki.raptorcs.com/wiki/PM8068))
+platform.
+
+PowerPC by default uses big endian. It is configurable at runtime, but the
+libraries included in coreboot-sdk are compiled for BE. This gives a perfect
+opportunity to validate that **coreboot works also for big endian platforms**.
+By now we confirmed that there are parts in CBFS and FMAP which need small
+fixes. The endianness of fields also needs to be properly documented for those
+structures, as they can be used both by a platform it is started on, as well as
+an application on a platform it is built on. Those two do not have to use the
+same endianness.
+
+The biggest achievement of this port would undoubtedly be bringing support for
+**native DDR4 initialisation and training**. Even though the details of
+accessing the hardware are platform-specific, the general process stays the
+same. One caveat might be that Talos II supports only RDIMM (registered, or
+buffered DIMM) and that is the only kind of memory sticks we are able to test,
+however even if it won't work out of the box for different DIMM flavours, it
+should give a good base to build on.
+
+## Current state of work
+
+We have successfully started coreboot on the platform:
+
+[![asciicast](https://asciinema.org/a/JQ1MaBSzGN1L1JcbgTX3G3kt6.svg)](https://asciinema.org/a/JQ1MaBSzGN1L1JcbgTX3G3kt6?speed=1)
+
+Actually, we were starting it much earlier than we knew it, because we used a
```

Comment:
...() than we got acknowledged with the platform?

Reply:
Maybe being more precise on the coreboot stage that was successful here would be advisable then saying coreboot was started (as per tweets and everywhere else)